fuse_optimizers/src/variable_stamp_index.cpp

```diff
@@ -47,19 +47,29 @@ namespace fuse_optimizers
 {
 ros::Time VariableStampIndex::currentStamp() const
 {
-  auto compare_stamps = [](const StampedMap::value_type& lhs, const StampedMap::value_type& rhs)
-  {
-    return lhs.second < rhs.second;
-  };
-  auto iter = std::max_element(stamped_index_.begin(), stamped_index_.end(), compare_stamps);
-  if (iter != stamped_index_.end())
-  {
-    return iter->second;
+  if(!unique_stamps_.empty()){
+    return *unique_stamps_.end();
```

Comment:
Doesn't this throw or segfault or something? The `end()` iterator is not dereferenceable.
```suggestion
    return *unique_stamps_.rbegin();
```